src/order/rel_iso.lean

```diff
@@ -307,6 +307,30 @@ protected theorem well_founded : ∀ (f : r ↪r s) (h : well_founded s), well_f
 protected theorem is_well_order : ∀ (f : r ↪r s) [is_well_order β s], is_well_order α r
 | f H := by exactI {wf := f.well_founded H.wf, ..f.is_strict_total_order'}
 
+/-- A relation embedding between the lift of a relation and the relation. -/
+@[simps] noncomputable def _root_.quotient.rel_embedding [s : setoid α] {r : α → α → Prop}
+  (H : ∀ a₁ a₂ b₁ b₂ : α, a₁ ≈ b₁ → a₂ ≈ b₂ → r a₁ a₂ = r b₁ b₂) : quotient.lift₂ r H ↪r r :=
+⟨embedding.quotient α, begin
+  refine λ x y, quotient.induction_on₂ x y (λ a b, _),
+  apply iff_iff_eq.2 (H _ _ _ _ _ _);
+  apply quotient.mk_out
+end⟩
+
+/-- A relation is well founded iff its lift to a quotient is. -/
+@[simp] theorem _root_.well_founded_lift₂_iff [s : setoid α] {r : α → α → Prop}
+  (H : ∀ a₁ a₂ b₁ b₂ : α, a₁ ≈ b₁ → a₂ ≈ b₂ → r a₁ a₂ = r b₁ b₂) :
+  well_founded (quotient.lift₂ r H) ↔ well_founded r :=
```

Comment:
You might want one for `quotient.lift_on₂' r H` too, which is trivial to prove.

Reply:
`quotient.lift_on₂'` is reducible and def-eq to `quotient.lift_on₂`, which itself is reducible and def-eq to `quotient.lift₂`. So I think just having this last version should be enough.

Reply:
Can you verify that `simp` can prove the primed version by itself?

Reply:
It can. And now that I think about it, `λ x y, quotient.lift_on₂' x y r H` is a very awkward spelling for `quotient.lift₂ r H`, so I doubt this will come up in practice.

Reply:
`quotient.lift_on₂'` is useful for when the setoid is not an instance, which is quite common. But if `simp` doesn't need the lemma about it here, then there's probably no issue.